go/libraries/events/event_flush.go

```diff
@@ -0,0 +1,145 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/golang/protobuf/proto"
+	eventsapi "github.com/liquidata-inc/dolt/go/gen/proto/dolt/services/eventsapi_v1alpha1"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
+	"github.com/liquidata-inc/dolt/go/libraries/utils/filesys"
+)
+
+// EventGrpcFlush parses dolt event logs sends the events to the events server
+type EventGrpcFlush struct {
+	em  *GrpcEmitter
+	fbp *FileBackedProc
+}
+
+// ErrEventsDataDir occurs when events are trying to be  flushed, but the events data directory
+// does not yet exist
+var ErrEventsDataDir = errors.New("unable to flush, events data directory does not exist")
+
+func GetGRPCEmitter(dEnv *env.DoltEnv) *GrpcEmitter {
+	host := dEnv.Config.GetStringOrDefault(env.MetricsHost, env.DefaultMetricsHost)
+	portStr := dEnv.Config.GetStringOrDefault(env.MetricsPort, env.DefaultMetricsPort)
+	insecureStr := dEnv.Config.GetStringOrDefault(env.MetricsInsecure, "false")
+
+	port, err := strconv.ParseUint(*portStr, 10, 16)
+
+	if err != nil {
+		log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid port.", env.MetricsPort, *portStr))
+		return nil
+	}
+
+	insecure, err := strconv.ParseBool(*insecureStr)
+
+	if err != nil {
+		log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid true/false value", env.MetricsInsecure, *insecureStr))
+	}
+
+	hostAndPort := fmt.Sprintf("%s:%d", *host, port)
+	conn, _ := dEnv.GrpcConnWithCreds(hostAndPort, insecure, nil)
+
+	return NewGrpcEmitter(conn)
+}
+
+// NewEventGrpcFlush creates a new EventGrpcFlush
+func NewEventGrpcFlush(fs filesys.Filesys, userHomeDir string, doltDir string, dEnv *env.DoltEnv) *EventGrpcFlush {
+
+	fbp := NewFileBackedProc(fs, userHomeDir, doltDir, MD5FileNamer)
+
+	if exists := fbp.EventsDirExists(); !exists {
+		panic(ErrEventsDataDir)
+	}
+
+	return &EventGrpcFlush{em: GetGRPCEmitter(dEnv), fbp: fbp}
+}
+
+func checkFilenameAndExt(data []byte, path string) (bool, error) {
+	filename := filepath.Base(path)
+	ext := filepath.Ext(filename)
+
+	if ext != evtDataExt {
+		return false, nil
+	}
+
+	md5FromFilename := filename[:len(filename)-len(ext)]
+	md5FromData := MD5StrUrl(data)
+
+	if md5FromFilename != md5FromData {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (egf *EventGrpcFlush) flush(path string, size int64, isDir bool) (stop bool) {
+	fs := egf.fbp.GetFileSys()
+
+	data, err := fs.ReadFile(path)
+	if err != nil {
+		log.Print(err)
+		return false
+	}
+
+	isFileValid, err := checkFilenameAndExt(data, path)
```

Comment:
if isFileValid is false, and err is nil that means the file is corrupted.  Retries on this file won't fix it, so we should go ahead and delete it.